content/blog/posts/2022-07-08-go-serverless-functions-with-ember-and-sst-on-aws.md

```diff
@@ -0,0 +1,200 @@
+
+We won't re-hash all the benefits of adding serverless functions alongside your frontend application, but we should mention that we think you can gain greater developer productivity from having them in your project. They can unblock frontend developers from relying on backend developers to write endpoints for some basic API functionality and you could write your entire backend as a set of serverless functions.
+
+Now, as we demonstrated in our last post on this subject, developing and deploying serverless functions to Vercel or Netlify is a breeze because of the tooling they provide. But, what if your project has constraints that require a deployment directly to AWS without the use of those other platforms?
+
+Thankfully, there's an answer to that question that isn't too involved. So today, we'll focus on deploying serverless functions directly to AWS using a framework called [SST](https://sst.dev/) that makes it trivial to build full-stack serverless apps.
+
+SST will handle generating the serverless functions, provide a way to run them locally, and also a method for deploying these functions, along with your frontend code, to AWS.
+
+![SST Homepage](/img/blog/serverlesss-functions-direct-to-aws-with-ember/sst-homepage.png)
+
+## Generating our SST Project
+
+Before we start you'll want to ensure you have an AWS account with the AWS CLI configured locally.
+
+By default, if you run `npx create-sst <your-app-name>`, SST provides a full-stack application, including an RDS or Dynamo database, and a GraphQL API as well - among other things. Those offerings are powerful and could be a solution for your project, but for the sake of simplicity and staying focused on only adding serverless functions as API Routes for our application, we'll use a more stripped-down [SST template](https://docs.sst.dev/packages/create-sst#--template) to limit what is installed.
+
+To get started we'll run `npx create-sst@latest --minimal <your-app-name>`. This will generate a new SST project that includes only the minimal set of things needed to form an SST app with serverless functions. You'll notice that we are presented with an option to choose a template:
+
+![Minimal SST Command and serverless function template choice](/img/blog/serverlesss-functions-direct-to-aws-with-ember/select-a-template-sst-minimal.png)
+
+We could choose the `javascript-starter` template if we wanted to create a serverless function written in Node.js. However, since we're presented with the option, let's do something different this time and choose `go-starter`. This will generate the project for us, but the serverless functions will be written in [Go lang](https://go.dev/) instead. Here's a look at the Go servless function that's generated:
```

Comment:
We could choose the `javascript-starter` template ~if we wanted~ to create a serverless function written in Node.js.